### DISCUSSION TITLE
How to overwrite a page from the filesystem when editing it in the annotation editor?

### DISCUSSION THREAD

**Discussion by twh270** [gildas-lormeau/SingleFile#795] [Q&A]
**Describe the bug**
After adding an annotation to an already-saved page, saving the annotated page saves directly to Downloads/ instead of following the configured file template.

**To Reproduce**
1. Save a page to e.g. Downloads\SavedPage.html
2. Open Downloads\SavedPage.html in the same or different tab. The URL should be file:///C:/Users/Me/Downloads/SavedPage.html
3. Right-Click the SingleFile toolbar icon and select "Annotate the page...". The URL should change to moz-extension:... and the page should show the SingleFile toolbar, with different-colored sticky notes etc.
4. Add a sticky note to the page. 
5. Save the page.
6. Navigate to the Downloads/ folder. The saved page will appear there.

**Expected behavior**
Expected that "save" would overwrite the existing saved page using the file template.

**Environment**
 - OS: Win10 Pro
 - Browser: Firefox
 - Version: 92.0.1

**Comment by gildas-lormeau** [comment]
You can achieve this by following these instructions:
 - Go to the options page of SingleFile
 - Click on the `+` button at the top right to create a new profile
 - Enter the name of the profile, e.g. `local pages` (or whatever else you prefer)
 - Make sure this profile is selected (check the name at the top right)
 - Open the section `File name`
 - Set `{url-last-segment}` as value for the setting `template`
 - Select the setting `file name conflict resolution > overwrite the existing file`
 - Open the section `Auto-settings rule`
 - Type `file:///` for the `URL` value in the first row of the table
 - Make sure `local pages` is the selected value for the `Profile` value
 - Click on the `+` button at the right the add the row in the table

Then, it should work as expected. Pages served from the web (i.e. `http` or `https` protocols) will follow the file name template defined in the `Default settings` profile. Pages served from the filesystem (i.e. with URLs containing `file:///`) will be overwritten thanks to the settings in the `File name` section of the `local pages` profile.

**Comment by gildas-lormeau** [comment]
I'm moving this issue in the [Discussions](https://github.com/gildas-lormeau/SingleFile/discussions) section.

**Comment by bogdan-h** [comment]
Hi Gildas,
I really need help here as I was never able to re-open a page saved with SingleFile (chrome) after annotating it ! This is one of the reasons I stopped using the extension. I have to admit this extension is fantastic, but I  cannot finish reading and annotating a page always in "one-try".

I enabled the option to open the page saved from SingleFile with annotations ... but it does not happen ! 
If there something that I'm missing ? 

Thank you in advance,
Bogdan.

**Reply by gildas-lormeau** [reply]
Hi @bogdan-h, if you're using Chrome or a browser based on Chromium make sure "Allow access to file URLs" is enabled the extension page, e.g. chrome://extensions/?id=mpiodijhokgodhhofbcjdecpffjipkle

**Reply by bogdan-h** [reply]
Thank you sooo much Gildas - that was the option that I was missing. Probably it was specified somewhere in the documentation .. but who reads the docs 😂.
Have a nice weekend and thank you again.

**Reply by gildas-lormeau** [reply]
I'm glad to know that it was the cause of the problem. FYI, It's documented in the known issues https://github.com/gildas-lormeau/SingleFile#known-issues.
Have a nice weekend too!
